### PR TITLE
rpc: remove duplicate solvable field from getaddressinfo

### PR DESCRIPTION
**Pull request by fanquake**
Also added optional to `iscompressed`.

### CODE REVIEW

**Comment by promag** [comment]
ACK, nice catch.

**Comment by fanquake** [comment]
Travis failure seems unrelated. Force pushed, added `optional` to `iscompressed`.

**Comment by laanwj** [comment]
Huh, how did we have it end up like this, a silent merge conflict?

**Comment by Empact** [comment]
Why is `iscompressed` optional? It's being added alongside `pubkey` afact, so they should both or neither be, right?

https://github.com/bitcoin/bitcoin/blob/fcb6694a9945d2a02f40587e18bd395ef64048e0/src/wallet/rpcwallet.cpp#L3499-L3508

**Comment by hebasto** [comment]
utACK a5db68c22ba03e661d20b1855e6a443019080690

**Comment by promag** [comment]
Should backport?

**Comment by fanquake** [comment]
@promag `getaddressinfo` in 0.17 doesn't have any `solvable` fields, so shouldn't be required.
https://github.com/bitcoin/bitcoin/blob/0a0b2ea11fde7becbdd1c04205336c3a39bffbc0/src/wallet/rpcwallet.cpp#L4149

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -3605,7 +3604,7 @@ UniValue getaddressinfo(const JSONRPCRequest& request)
             "  \"sigsrequired\" : xxxxx        (numeric, optional) Number of signatures required to spend multisig output (only if \"script\" is \"multisig\")\n"
             "  \"pubkey\" : \"publickeyhex\",    (string, optional) The hex value of the raw public key, for single-key addresses (possibly embedded in P2SH or P2WSH)\n"
             "  \"embedded\" : {...},           (object, optional) Information about the address embedded in P2SH or P2WSH, if relevant and known. It includes all getaddressinfo output fields for the embedded address, excluding metadata (\"timestamp\", \"hdkeypath\", \"hdseedid\") and relation to the wallet (\"ismine\", \"iswatchonly\").\n"
-            "  \"iscompressed\" : true|false,  (boolean) If the address is compressed\n"
+            "  \"iscompressed\" : true|false,  (boolean, optional) If the address is compressed\n"
```
Comment (MarcoFalke) [inline src/wallet/rpcwallet.cpp]: Shouldn't this say pubkey?

```suggestion
            "  \"iscompressed\" : true|false,  (boolean, optional) If the pubkey is compressed\n"
```

**Comment by MarcoFalke** [comment]
ACK

**Comment by achow101** [comment]
utACK a5db68c22ba03e661d20b1855e6a443019080690

**Comment by IlyasRidhuan** [comment]
utAck

**Pull request merged**